include/shared/config/conf.hpp

```diff
@@ -0,0 +1,64 @@
+#ifndef CONF_H
+#define CONF_H
+
+#include <map>
+#include <string>
+#include <iostream>
+
+#define NEG -1 // Negative value when it's false
+
+enum Configuration {
+   SERVER_IP,
+   SERVER_PORT,
+   CLIENT_IP,
+   CLIENT_PORT,
+   LAST
+};
+
+class Config {
+   static Config* m_instance;
+   std::string const m_base_conf = "../config/config.conf";
+   std::string m_user_conf;
+   std::string m_user_dir;
+
+   bool           compare_files();
+
+public:
+   Config();
```

Comment:
If the intention is that `Config` is a singleton then the constructor has to be private.  Otherwise, you could just instantiate Config objects anywhere in the codebase.